feat: add tracing

lucasfernog-crabnebula opened the pull request:
<!--
Update "[ ]" to "[x]" to check a box

Please make sure to read the Pull Request Guidelines: https://github.com/tauri-apps/tauri/blob/dev/.github/CONTRIBUTING.md#pull-request-guidelines
-->

### What kind of change does this PR introduce?
<!-- Check at least one. If you are introducing a new binding, you must reference an issue where this binding has been proposed, discussed and approved by the maintainers. -->

- [ ] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?
<!-- If yes, please describe the impact and migration path for existing applications in an attached issue. -->

- [ ] Yes, and the changes were approved in issue #___
- [ ] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### Other information

lucasfernog-crabnebula committed:
window tracing

lucasfernog-crabnebula committed:
fix run never resolving all spans

lucasfernog-crabnebula committed:
fix draw not entered

lucasfernog-crabnebula committed:
change level

lucasfernog-crabnebula committed:
feat(core): Manager::emit_filter and optimize serialization (#7512)

Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

lucasfernog-crabnebula committed:
event spans

lucasfernog-crabnebula committed:
lint & fix tests

CrabNejonas commented:
I like the name spacing! Just checked with the `tokio` repo and they use `::` (instead of `.`) for namespacing, should we do that as well for consistency?

lucasfernog-crabnebula commented:
Of course you like the namespace, I copied it from your PR 😂 let's do what tokio is doing.

CrabNejonas commented:
> Of course you like the namespace, I copied it from your PR 😂 let's do what tokio is doing.

oh lol, I just looked at my linear issue which doesn't have that but hey 😂 you're right it would explain that

lucasfernog-crabnebula committed:
change eval to run sync

lucasfernog-crabnebula committed:
fix instrument

lucasfernog-crabnebula committed:
update wry

lucasfernog-crabnebula committed:
change separator

CrabNejonas commented on core/tauri-macros/src/command/wrapper.rs:
```diff
@@ -211,7 +228,17 @@ pub fn wrapper(attributes: TokenStream, item: TokenStream) -> TokenStream {
+          let #root::ipc::Invoke { id: #id, message: #message, resolver: #resolver } = $invoke;
+
+          let _span = tracing::debug_span!(
+            "ipc::request::handler",
```
Isn't this the top-level span for ipc requests? If so I would just use `ipc::request` for this and keep the `ipc::request::handler` for what's called `ipc::request::run` right now

```suggestion
            "ipc::request",
```

lucasfernog-crabnebula replied:
it's not the top level, there's more logic in window itself before it gets here (plus right now the top-level span is in wry).

CrabNejonas commented on core/tauri-macros/src/command/wrapper.rs:
```diff
@@ -234,14 +261,22 @@ fn body_async(
     quote! {
+      use tracing::Instrument;
+
+      let span = tracing::debug_span!("ipc::request::run", id = #id.0);
```
```suggestion
      let span = tracing::debug_span!("ipc::request::handler", id = #id.0);
```

See previous comment

CrabNejonas commented on core/tauri-macros/src/command/wrapper.rs:
```diff
@@ -267,6 +306,7 @@ fn body_blocking(
   });
 
   Ok(quote! {
+    let _span = tracing::debug_span!("ipc::request::run", id = #id.0).entered();
```
```suggestion
    let _span = tracing::debug_span!("ipc::request", id = #id.0).entered();
```

CrabNejonas commented on core/tauri-runtime-wry/src/lib.rs:
```diff
@@ -221,11 +221,35 @@ impl<T: UserEvent> Context<T> {
 }
 
+#[derive(Debug, Clone, Default)]
+pub struct ActiveTraceSpanStore(Rc<RefCell<Vec<ActiveTracingSpan>>>);
```
What is this used for?

lucasfernog-crabnebula replied:
to track how much time it took to draw the window, so we keep the span alive for a while

CrabNejonas commented on core/tauri-runtime-wry/src/lib.rs:
```diff
@@ -2565,6 +2606,11 @@ fn create_webview<T: UserEvent, F: Fn(RawWindow) + Send + 'static>(
     ..
   } = pending;
 
+  let _webview_create_span = tracing::debug_span!("wry::webview::create").entered();
+  let window_draw_span = tracing::debug_span!("wry::window::draw").entered();
+  let window_create_span =
+    tracing::debug_span!(parent: &window_draw_span, "wry::window::create").entered();
```
Do you think these might be better suited to live in the `wry` repo now that we have forked it too?  (in which case we wouldn't need the `wry::` prefix that we can derive from the span metadata)

lucasfernog-crabnebula replied:
I don't really want to bloat wry with a bunch of tracing calls, but whatever we all decide works for me. cc @Wu-Yu-Wei

CrabNejonas replied:
I think adding tracing to wry makes sense too (and we'll get the request at some point anyway) as even users that use wry just as a library would want to get perf info

CrabNejonas commented on core/tauri/src/app.rs:
```diff
@@ -356,26 +356,14 @@ impl<R: Runtime> AppHandle<R> {
-      .lock()
-      .unwrap()
-      .register(plugin);
+  #[tracing::instrument(name = "app.plugin.register", skip(plugin), fields(name = plugin.name()))]
```
```suggestion
  #[tracing::instrument(name = "app::plugin::register", skip(plugin), fields(name = plugin.name()))]
```

CrabNejonas commented on core/tauri/src/app.rs:
```diff
@@ -808,6 +796,7 @@ shared_app_impl!(App<R>);
 shared_app_impl!(AppHandle<R>);
 
 impl<R: Runtime> App<R> {
+  #[tracing::instrument(name = "app.core_plugins.register")]
```
```suggestion
  #[tracing::instrument(name = "app::core_plugins::register")]
```

CrabNejonas commented on core/tauri/src/app.rs:
````diff
@@ -940,6 +929,7 @@ impl<R: Runtime> App<R> {
   /// }
   /// ```
   #[cfg(desktop)]
+  #[tracing::instrument(name = "app.run_iteration")]
````
```suggestion
  #[tracing::instrument(name = "app::run_iteration")]
```

CrabNejonas commented on core/tauri/src/app.rs:
```diff
@@ -1460,6 +1450,7 @@ impl<R: Runtime> Builder<R> {
 
   /// Builds the application.
   #[allow(clippy::type_complexity)]
+  #[tracing::instrument(name = "app.build", skip(self))]
```
```suggestion
  #[tracing::instrument(name = "app::build", skip(self))]
```

CrabNejonas commented on core/tauri/src/app.rs:
```diff
@@ -1686,6 +1677,7 @@ unsafe impl<R: Runtime> HasRawDisplayHandle for App<R> {
   }
 }
 
+#[tracing::instrument(name = "app.setup")]
```
```suggestion
#[tracing::instrument(name = "app::setup")]
```

CrabNejonas commented on core/tauri/src/app.rs:
```diff
@@ -1739,6 +1731,7 @@ fn setup<R: Runtime>(app: &mut App<R>) -> crate::Result<()> {
   Ok(())
 }
 
+#[tracing::instrument(name = "app.event_loop", skip(app_handle, callback, manager))]
```
```suggestion
#[tracing::instrument(name = "app::event_loop", skip(app_handle, callback, manager))]
```

CrabNejonas commented on core/tauri/src/ipc/protocol.rs:
```diff
@@ -275,6 +326,7 @@ fn handle_ipc_message<R: Runtime>(message: String, manager: &WindowManager<R>, l
         );
       }
       Err(e) => {
+        tracing::trace!("ipc.request.error {}", e);
```
```suggestion
        tracing::trace!("ipc::request::error {}", e);
```

CrabNejonas commented on core/tauri/src/ipc/protocol.rs:
```diff
@@ -42,11 +47,24 @@ pub fn get<R: Runtime>(manager: WindowManager<R>, label: String) -> UriSchemePro
                 Box::new(move |_window, _cmd, response, _callback, _error| {
+                  let _span = tracing::trace_span!(
+                    parent: &request_span,
+                    "ipc::request::respond",
```
This span ID is way to similar to `ipc::request::response`, It is also duplicating the data sent by the next span

CrabNejonas commented on core/tauri/src/lib.rs:
````diff
@@ -568,10 +568,32 @@ pub trait Manager<R: Runtime>: sealed::ManagerBase<R> {
   ///   app.emit_all("synchronized", ());
   /// }
   /// ```
+  #[tracing::instrument("app.emit.all", skip(self, payload))]
````
```suggestion
  #[tracing::instrument("app::emit_all", skip(self, payload))]
```

CrabNejonas commented on core/tauri/src/lib.rs:
````diff
@@ -568,10 +568,32 @@ pub trait Manager<R: Runtime>: sealed::ManagerBase<R> {
+  ///   app.emit_filter("synchronized", (), |w| w.label().starts_with("foo-"));
+  /// }
+  /// ```
+  #[tracing::instrument("app.emit.filter", skip(self, payload, filter))]
````
```suggestion
  #[tracing::instrument("app::emit_filter", skip(self, payload, filter))]
```

CrabNejonas commented on core/tauri/src/lib.rs:
````diff
@@ -587,6 +609,7 @@ pub trait Manager<R: Runtime>: sealed::ManagerBase<R> {
   ///   }
   /// }
   /// ```
+  #[tracing::instrument("app.emit.to", skip(self, payload))]
````
```suggestion
  #[tracing::instrument("app::emit_to", skip(self, payload))]
```

CrabNejonas commented on core/tauri/src/lib.rs:
````diff
@@ -651,6 +674,7 @@ pub trait Manager<R: Runtime>: sealed::ManagerBase<R> {
   ///   }
   /// }
   /// ```
+  #[tracing::instrument("app.emit.rust", skip(self))]
````
```suggestion
  #[tracing::instrument("app::emit_rust", skip(self))]
```

I don't like that the span name differs from the method name so much

CrabNejonas commented on core/tauri/src/plugin.rs:
```diff
@@ -646,10 +655,25 @@ impl<R: Runtime> PluginStore<R> {
   }
 }
+
+#[tracing::instrument(name = "plugin.hooks.initialize", skip(plugin), fields(name = plugin.name()))]
```
```suggestion
#[tracing::instrument(name = "plugin::hooks::initialize", skip(plugin), fields(name = plugin.name()))]
```

CrabNejonas commented on core/tauri/src/window.rs:
```diff
@@ -371,6 +391,7 @@ impl<'a, R: Runtime> WindowBuilder<'a, R> {
   }
 
   /// Creates a new webview window.
+  #[tracing::instrument(name = "window.create")]
```
```suggestion
  #[tracing::instrument(name = "window::create")]
```

CrabNejonas commented on core/tauri/src/window.rs:
```diff
@@ -948,6 +971,7 @@ impl<R: Runtime> PartialEq for Window<R> {
 }
 
 impl<R: Runtime> Manager<R> for Window<R> {
+  #[tracing::instrument("window.emit.to", skip(self, payload))]
```
```suggestion
  #[tracing::instrument("window::emit_to", skip(self, payload))]
```

CrabNejonas commented on core/tauri/src/window.rs:
```diff
@@ -959,12 +983,14 @@ impl<R: Runtime> Manager<R> for Window<R> {
       .emit_filter(event, Some(self.label()), payload, |w| label == w.label())
   }
 
+  #[tracing::instrument("window.emit.all", skip(self, payload))]
```
```suggestion
  #[tracing::instrument("window::emit_all", skip(self, payload))]
```

CrabNejonas commented on core/tauri/src/window.rs:
```diff
@@ -2418,18 +2457,22 @@ impl<R: Runtime> Window<R> {
-    payload: S,
-  ) -> crate::Result<()> {
+  #[tracing::instrument(
+    "window.emit",
```
```suggestion
    "window::emit",
```

CrabNejonas commented:
Okay so general comments:

1. I think it makes sense to pull out the span IDs into constants if possible (deduplicate them from the binary and help with maintaining them later)
2. I can't check this without a proper span visualisation but I think in places you're not dropping spans guards before entering the next (doesn't matter right now though we can change that as we develop devtools)

lucasfernog-crabnebula committed:
Update core/tauri/src/plugin.rs

Co-authored-by: Jonas Kruckenberg <118265418+CrabNejonas@users.noreply.github.com>

lucasfernog-crabnebula committed:
Update core/tauri/src/window.rs

Co-authored-by: Jonas Kruckenberg <118265418+CrabNejonas@users.noreply.github.com>

lucasfernog-crabnebula committed:
Update core/tauri/src/window.rs

Co-authored-by: Jonas Kruckenberg <118265418+CrabNejonas@users.noreply.github.com>

lucasfernog-crabnebula committed:
Update core/tauri/src/window.rs

Co-authored-by: Jonas Kruckenberg <118265418+CrabNejonas@users.noreply.github.com>

lucasfernog-crabnebula committed:
Update core/tauri/src/window.rs

Co-authored-by: Jonas Kruckenberg <118265418+CrabNejonas@users.noreply.github.com>

lucasfernog-crabnebula committed:
instrument separators

lucasfernog-crabnebula commented:
@CrabNejonas we do have this awesome span visualisation tool called devtools 😂

lucasfernog-crabnebula commented:
I guess we should have a central module for span names right?

CrabNejonas commented:
> I guess we should have a central module for span names right?

Yes or go straight with something like https://github.com/crabnebula-dev/devtools/pull/32

logankeenan committed:
fix: remove mut self from set_native_icon signature (#7866)

* fix set_native_icon compile error

* Update core/tauri/src/menu/icon.rs

* Update core/tauri/src/menu/icon.rs

* Create set_native_icon_signature.md

* Update .changes/set_native_icon_signature.md

amrbashir committed:
docs: use `http::*` qualified import instead of an alias (#7873)

amrbashir committed:
refactor!: remove `tauri::api` module (#7874)

* refactor!: remove `tauri::api` module

ref: https://github.com/tauri-apps/tauri/issues/7756

* change file

* fix builds

amrbashir committed:
refactor!: changed `Env.args` to `Env.args_os` and use `OsString` instead of `String` (#7876)

ref: https://github.com/tauri-apps/tauri/issues/7756

tr3ysmith committed:
fix: rpath missing from app, closes #7710 (#7773)

Co-authored-by: Lucas Nogueira <lucas@tauri.app>
Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.studio>
fix: codesign doesn't sign frameworks or sidecar, closes #7690 (#7774)

FabianLars committed:
docs: change `icon` help output to ask for 1024px source icons, closes #7886 (#7896)

FabianLars committed:
feat(core): Allow http origin on Windows, fixes: #3007 (#7645)

Co-authored-by: Lucas Nogueira <118899497+lucasfernog-crabnebula@users.noreply.github.com>
Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

FabianLars committed:
fix(cli): help output mixed up for private-key and private-key-path (#7901)

lucasfernog committed:
fix(cli): icons not added to the Android project (#7902)

Benji377 committed:
fix: Returns a warning instead of panicking if an AppImage is not mounted, closes #7736 (#7843)

Co-authored-by: Lucas Nogueira <lucas@tauri.app>
Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.studio>

lucasfernog committed:
fix: Returns a warning instead of panicking if an AppImage is not mounted, closes #7736 (#7912)

Co-authored-by: Benjamin Demetz <50681275+Benji377@users.noreply.github.com>

lucasfernog committed:
fix(ci): downgrade async-executor to 1.5.2 (#7911)

github-actions[bot] committed:
Apply Version Updates From Current Changes (v1) (#7261)

Co-authored-by: lucasfernog <lucasfernog@users.noreply.github.com>

lucasfernog committed:
fix(ci): update tauri-driver's Cargo.lock file on version bump (#7915)

lucasfernog committed:
hotfix(build): bump tauri-build to 1.5 (#7918)

github-actions[bot] committed:
Apply Version Updates From Current Changes (v1) (#7924)

Co-authored-by: amrbashir <amrbashir@users.noreply.github.com>

lucasfernog committed:
fix(core): toggle devtools via global shortcut (#7937)

* fix(core): toggle devtools via global shortcut

* Update .changes/fix-toggle-devtools.md

pewsheen committed:
fix(macos): fix notraytool's apple-id option name, close #7917 (#7934)

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>

github-actions[bot] committed:
Apply Version Updates From Current Changes (v1) (#7938)

Co-authored-by: lucasfernog <lucasfernog@users.noreply.github.com>

amrbashir committed:
refactor!: consistent naming in `tauri::scope` module (#7944)

amrbashir committed:
refactor!: renamed `TrayIconBuilder/TrayIcon::on_tray_event` to `TrayIconBuilder/TrayIcon::on_tray_icon_event` (#7943)

martinfrances107 committed:
fix: Cleared warnings seen while running ```cargo doc``` (#7862)

lucasfernog committed:
fix(core): update scope usage following #7944 (#7946)

amrbashir committed:
feat: add `id` option for tray icon in config file (#7871)

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.studio>

lucasfernog committed:
fix(cli): deadlock regression from #7802 (#7948)

FabianLars committed:
fix(bundler): Use escaped installer path when starting the updater setup, fixes #7931 (#7956)

* fix(bundler): Use escaped installer path when starting the updater setup, fixes #7931

* fun ci stuff

* who needs lockfiles anyway

* stahp

* please let it be the last one. i want to leave

amrbashir committed:
chore: change `tauri-mobile` to `cargo-mobile2` (#7953)

lucasfernog committed:
fix(bundler): team ID is now required for notarytool via app password (#7972)

lucasfernog committed:
feat(cli): use devicetl to connect with iOS 17 devices (#7971)

github-actions[bot] committed:
Apply Version Updates From Current Changes (#7823)

Co-authored-by: lucasfernog <lucasfernog@users.noreply.github.com>

github-actions[bot] committed:
Apply Version Updates From Current Changes (v1) (#7960)

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>
Co-authored-by: lucasfernog <lucasfernog@users.noreply.github.com>
fix(bundler): team ID is now required for notarytool via app password (#7972)
fix(bundler): team ID is now required for notarytool via app password (#7972)"

zh217 committed:
fix: truncate file when not appending in `writeFile`, closes #7973 (#7982)

* Fix https://github.com/tauri-apps/tauri/issues/7973

* Fix https://github.com/tauri-apps/tauri/issues/7973

* Fix https://github.com/tauri-apps/tauri/issues/7973

i-c-b committed:
fix(bundler): WebView2 offline installer GUID changes (#7998)

amrbashir committed:
refactor(cli): enhance plugin subcommand, closes #7749 (#7990)

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>

lucasfernog committed:
fix(ci): properly detect @tauri-apps/cli release id from covector output (#8004)

github-actions[bot] committed:
Apply Version Updates From Current Changes (v1) (#7988)

Co-authored-by: lucasfernog <lucasfernog@users.noreply.github.com>

lucasfernog committed:
feat(core): include mobile on docs.rs targets (#8006)

CaiJingLong committed:
fix(bundler): read proxy when downloading resources. (#8012)

* fix(bundler): read http_proxy env when downloading resources.

Signed-off-by: CaiJingLong <cjl_spy@163.com>

* Update .changes/add-proxy-for-nsis-download.md

* Update add-proxy-for-nsis-download.md

* Update tooling/bundler/src/bundle/windows/util.rs

---------

Signed-off-by: CaiJingLong <cjl_spy@163.com>

olivierlemasle committed:
fix(bundler): In .deb packages, set uid=0 for all files (#7980)

amrbashir committed:
chore: update tauri-cli lock file (#8015)

amrbashir committed:
chore: rename webview2 offline installer guid consts for more clarity (#8001)

amrbashir committed:
docs: document macOS restrictions on `Menu::add_item` (#7983)

amrbashir committed:
refactor: enhance the descriptions of CLI commands, closes #7572 (#7952)

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>
Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

amrbashir committed:
refactor!: remove uuid and rand dependencies where applicable (#7939)

* refactor: remove uuid and rand dependencies where applicable

ref: https://github.com/tauri-apps/tauri/issues/7756

* replace rand with getrandom

* change files

* InnerListeners private

* revert listeners_object_name [skip ci]

* default for next_event_id

* remove raw listen function

* fix event system

* Apply suggestions from code review [skip ci]

* update names [skip ci]

---------

Co-authored-by: Lucas Nogueira <lucas@tauri.studio>
Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>

dependabot[bot] committed:
chore(deps): bump @babel/traverse from 7.17.0 to 7.23.2 in /tooling/cli/node (#8026)

Bumps [@babel/traverse](https://github.com/babel/babel/tree/HEAD/packages/babel-traverse) from 7.17.0 to 7.23.2.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.23.2/packages/babel-traverse)

---
updated-dependencies:
- dependency-name: "@babel/traverse"
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

amrbashir committed:
feat: move `window` plugin back to core (#8007)

Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

amrbashir committed:
refactor!: consistent environment variables (#7949)

Co-authored-by: Fabian-Lars <fabianlars@fabianlars.de>
Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

dependabot[bot] committed:
chore(deps): bump undici from 5.19.1 to 5.26.3 in /examples/api (#8024)

Bumps [undici](https://github.com/nodejs/undici) from 5.19.1 to 5.26.3.
- [Release notes](https://github.com/nodejs/undici/releases)
- [Commits](https://github.com/nodejs/undici/compare/v5.19.1...v5.26.3)

---
updated-dependencies:
- dependency-name: undici
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Amr Bashir <amr.bashir2015@gmail.com>

amrbashir committed:
feat(api): add mock for `convertFileSrc`, closes #7935 (#7961)

* feat(api): add mock for `convertFileSrc`, closes #7935

* fix lint

* Update tooling/api/src/mocks.ts

* fmt

amrbashir committed:
refactor(core&api)!: hide internal functions and reuse them in api.js & rename `tauri` module to `primitives` (#7942)

Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

lucasfernog committed:
fix(core): compile on 32-bit targets (#8032)

amrbashir committed:
chore: downgrade bump to patch (#8033)

lucasfernog committed:
chore: bump MSRV to 1.70 (#8031)

amrbashir committed:
feat(windows): add tabbed effect (#7794)

Co-authored-by: Lucas Nogueira <lucas@tauri.studio>
Co-authored-by: Lucas Nogueira <lucas@tauri.app>

github-actions[bot] committed:
Apply Version Updates From Current Changes (v1) (#8022)

Co-authored-by: lucasfernog <lucasfernog@users.noreply.github.com>
Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

amrbashir committed:
refactor(cli): cleanup info command (#7204)

Co-authored-by: martin frances <martinfrances107@hotmail.com>
Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

amrbashir committed:
refactor!: remove unnecessary error enum and result type aliases (#7875)

Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

amrbashir committed:
refactor(core/plugin)!: inline result type alias (#8037)

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>

amrbashir committed:
feat: move app plugin back to core (#8039)

Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

amrbashir committed:
refactor: change path and event plugins to follow the same convention as `window` (#8040)

pewsheen committed:
feat: set application progress bar, close #7999 (#8009)

Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

aelew committed:
feat(core): expose `always_on_bottom`, closes #7847 (#7933)

Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

martinfrances107 committed:
fix(core): Clippy fixes. (#7869)

Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

renovate[bot] committed:
chore(deps) Update Tauri CLI (1.x) (#7641)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

dependabot[bot] committed:
chore(deps): bump postcss from 8.4.24 to 8.4.31 in /examples/api (#7968)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Amr Bashir <amr.bashir2015@gmail.com>
Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

renovate[bot] committed:
chore(deps) Update Tauri API Definitions (dev) (major) (#7638)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

lucasfernog committed:
fix(cli): duplicated newlines on child process output (#8042)

renovate[bot] committed:
chore(deps) Update Rust crate log to 0.4.20 (dev) (#7629)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps) Update Tauri Bundler (1.x) (#7639)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps) Update Tauri API Definitions (dev) (#7633)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps) Update Tauri Utils (dev) (#7632)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>

lucasfernog committed:
refactor: change prefix of hook env vars to TAURI_ENV_, closes #8054 (#8057)

olivierlemasle committed:
fix: improve French translations for nsis installer (cherry-pick #7523 on 1.x) (#8058)

Co-authored-by: Stanislas Michalak <stanislas-m@users.noreply.github.com>

renovate[bot] committed:
chore(deps) Update Tauri CLI (dev) (#7630)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

lucasfernog committed:
feat(cli): allow rotating updater private key (#8059)

lucasfernog committed:
feat(bundler): enhance notarization error message (#7974)

github-actions[bot] committed:
Apply Version Updates From Current Changes (#8005)

Co-authored-by: lucasfernog <lucasfernog@users.noreply.github.com>
Co-authored-by: Lucas Nogueira <lucas@tauri.app>

github-actions[bot] committed:
Apply Version Updates From Current Changes (v1) (#8062)

Co-authored-by: lucasfernog <lucasfernog@users.noreply.github.com>

veldhaenchen committed:
feat(lang) added german language (#8061)

lucasfernog committed:
fix(core): prevent iOS crash on invalid plugin response JSON (#8049)

renovate[bot] committed:
chore(deps) Update Tauri Codegen (1.x) (#7576)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Lucas Nogueira <lucas@tauri.app>

renovate[bot] committed:
chore(deps) Update Tauri Utils (dev) (#8050)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps) Update Tauri Bundler (dev) (#8047)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps) Update Rust crate json-patch to 1.2 (dev) (#8043)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps) Update Tauri CLI (1.x) (#8051)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Lucas Nogueira <lucas@tauri.app>

lucasfernog committed:
Merge branch '1.x' into dev

G07cha committed:
fix(api): do nothing in clearMocks if __TAURI_INTERNALS__ is not defined (#8071)

olivierlemasle committed:
fix(examples): use __TAURI__.primitives (#8077)

Closes #8073

renovate[bot] committed:
chore(deps) Update Tauri Build (dev) (#8044)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

olivierlemasle committed:
fix(ci): Fix bench job (#8045)

Co-authored-by: Lucas Nogueira <lucas@tauri.app>

dependabot[bot] committed:
chore(deps): bump rustix from 0.37.23 to 0.37.26 in /tooling/bench (#8070)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

lucasfernog committed:
feat(plugins): typed invoke arguments for mobile plugins (#8076)

amrbashir committed:
refactor: enhance event system rust apis (#7996)

Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

FabianLars committed:
fix(deps): Downgrade minisign to 0.7.3 again (#8082)

amrbashir committed:
build(api.js): simplify output JS files (#8025)

Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

lucasfernog committed:
fix: e2e test using invalid updater private keys (#8086)

github-actions[bot] committed:
Apply Version Updates From Current Changes (v1) (#8065)

Co-authored-by: lucasfernog <lucasfernog@users.noreply.github.com>

dklassic committed:
chore(deps): upgrade gtk to 0.18 (#8084)

Co-authored-by: Lucas Nogueira <lucas@tauri.app>

amrbashir committed:
chore(deps): update windows-rs to 0.51 and webview2-com to 0.27 (#8035)

Co-authored-by: Lucas Nogueira <lucas@tauri.app>

lucasfernog-crabnebula committed:
remove on_event trace

lucasfernog committed:
feat(cli): add unit test for empty updater private key password (#8096)

lucasfernog committed:
fix(api): adjust documentation for mockConvertFileSrc (#8111)

renovate[bot] committed:
chore(deps) Update Rust crate libflate to v2 (1.x) (#8106)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps) Update Tauri API Definitions (1.x) (#8100)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps) Update dependency @types/node to v20.8.8 (dev) (#8098)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

olivierlemasle committed:
fix(cross): Fix docker image and cross setup (#8094)

olivierlemasle committed:
chore(ci): Replace unmaintained actions-rs GH actions, closes #8078 (#8093)

Co-authored-by: Lucas Nogueira <lucas@tauri.app>

lucasfernog committed:
chore(renovate): ignore cargo_toml and minisign (#8118)

lucasfernog committed:
chore: update plugins-workspace branch usage from next to v2 (#8120)

lucasfernog committed:
refactor(core): implement on_page_load event using wry hook (#8116)

lucasfernog committed:
fix(docs): set the --cfg docsrs arg for docs.rs build (#8123)

FabianLars committed:
feat(android): Add onResume and onPause to android plugins (#8092)

Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

github-actions[bot] committed:
Apply Version Updates From Current Changes (#8064)

Co-authored-by: lucasfernog <lucasfernog@users.noreply.github.com>

FabianLars committed:
fix(api): include tslib dist in api package (#8130)

github-actions[bot] committed:
Apply Version Updates From Current Changes (#8131)

Co-authored-by: lucasfernog <lucasfernog@users.noreply.github.com>

lucasfernog committed:
feat(cli): transform paths into relative for the mobile IDE script (#8128)

FabianLars committed:
fix(build): always invoke resource compiler on windows, fixes #8164 (#8168)

* fix(build): Always invoke resource compiler on windows, fixes #8164

* move all the stuff out of the version check

* check for icon path when setting the icon

ShoreCN committed:
refactor: Optimize Simplified Chinese language content (#8137)

i-c-b committed:
enhance(docs): Document Windows-specific icon behaviour (#8153)

fetzsav committed:
[feat (issue #6389)] make tauri icon support SVGs (#6444)

Co-authored-by: Fetzer <fetz@fetzverse.com>
Co-authored-by: Lucas Nogueira <lucas@tauri.studio>
Co-authored-by: FabianLars <fabianlars@fabianlars.de>

lucasfernog committed:
refactor(core): break manager struct into smaller ones (#8124)

lucasfernog committed:
fix(core): global events regression from #7996 closes #8146 (#8147)

moka-ayumu committed:
fix(bundler): modify version in output file name when bundling wix (#8182)

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>

renovate[bot] committed:
chore(deps) Update Rust crate cargo_toml to 0.17 (dev) (#8154)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps) Update Tauri API Definitions (dev) (#8151)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps) Update Rust crate tempfile to 3.8.1 (dev) (#8150)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

lucasfernog committed:
fix(cli): invalid plugin template (#8161)

vasfvitor committed:
fix: (cli) format plugin string to snake case, closes #8017 (#8149)

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>

renovate[bot] committed:
chore(deps) Update Tauri CLI (dev) (#8097)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

renovate[bot] committed:
chore(deps) Update Tauri Core (dev) (#7631)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

olivierlemasle committed:
fix(cli): Fixes errors on command output, closes #8110 (#8162)

Fixes #8110

renovate[bot] committed:
chore(deps) Update Rust crate syn to v2 (dev) (#7634)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

ptazithos committed:
chore(deps): replace deprecated  rollup-plugin-terser with @rollup/plugin-terser (#8194)

MGough committed:
fix(api): avoid crashing in `clearMocks`, closes #8179 (#8219)

* fix(api): do nothing in clearMocks if __TAURI_INTERNALS__ is not defined (fix: #8179)

Prevents tests that didn't define any mocks from failing in `afterEach` test hook.

* Update fix-clearmocks.md

* check for indvidiual properties instead

lucasfernog-crabnebula committed:
skip all arguments on App::build tracing

lucasfernog committed:
feat(cli): merge user-defined plist with the iOS plist file (#8200)

i-c-b committed:
Fix file-drop event payload (#8204)

Co-authored-by: Lucas Nogueira <lucas@tauri.app>

renovate[bot] committed:
chore(deps) Update Rust crate tempfile to 3.8.1 (1.x) (#8188)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps) Update Tauri API Definitions (dev) (#8187)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps) Update dependency eslint to v8.53.0 (1.x) (#8189)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

lucasfernog committed:
fix(core): docs.rs build failing for macOS (#8095)

lucasfernog committed:
fix(cli): IDE failing to read CLI options on `build --open` commands (#8202)

lucasfernog committed:
fix(cli): add support to Xcode's archive (#8209)

renovate[bot] committed:
chore(deps) Update Tauri CLI (1.x) (#8193)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Lucas Nogueira <lucas@tauri.app>

renovate[bot] committed:
chore(deps) Update Tauri Utils (1.x) (#8105)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Lucas Nogueira <lucas@tauri.app>
Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

renovate[bot] committed:
chore(deps) Update Tauri Build (1.x) (#8102)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Lucas Nogueira <lucas@tauri.app>

FabianLars committed:
fix(bundler/nsis): calculate estimated size on build system (#8233)

* fix(bundler): Fix nsis installer taking longer than expected to install resources

* create dir structure for resources before extracting files

* calculate size in rust on the build system

* i'm sorry clippy, i programmed in unholy languages where += wasn't a thing so i forgot it exists in rust...

* i'm a better clippy than clippy🌚

amrbashir committed:
fix(updater): Use escaped installer path when starting the updater msi (#8234)

* fix(updater): Use escaped installer path when starting the updater msi

Continuation of https://github.com/tauri-apps/tauri/pull/7956

* fix build

amrbashir committed:
feat: add tray and menu javascript APIs and events, closes #6617 (#7709)

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>
Co-authored-by: Fabian-Lars <fabianlars@fabianlars.de>
Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

amrbashir committed:
feat: migrate manual implementation to the light `windows-version` crate (#8243)

amrbashir committed:
feat: migrate manual implementation to the light `windows-version` crate (#8242)

amrbashir committed:
ci: fix parsing covector ouput (#8240)

renovate[bot] committed:
chore(deps) Update Rust crate handlebars to 4.5 (1.x) (#8231)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps) Update Tauri Core (dev) (#8227)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

FabianLars committed:
fix(bundler/nsis): remove empty resources folders on uninstall (#8263)

* fix(bundler/nsis): remove empty resources folders on uninstall

* make clippy happy for once

github-actions[bot] committed:
Apply Version Updates From Current Changes (#8140)

Co-authored-by: lucasfernog <lucasfernog@users.noreply.github.com>
Co-authored-by: Lucas Nogueira <lucas@tauri.app>

amrbashir committed:
ci: fix parsing covector ouput (#8241)

* ci: fix parsing covector ouput

* delete parse-output.js

---------

Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

FabianLars committed:
refactor(api)!: Rename `primitives` mod to `core` (#8273)

amrbashir committed:
perf: remove extranous call in `StateManager::get` (#8272)

amrbashir committed:
fix(api): add top-level `main`, `module` and `types` fields (#8267)

amrbashir committed:
fix(api): add top-level `main`, `module` and `types` fields (#8268)

amrbashir committed:
refactor(tauri-utils)!: remove `windows_version` and `is_windows_7` (#8271)

Co-authored-by: Lucas Nogueira <lucas@tauri.app>

Andreybest committed:
feat(bundler): add dmg settings, closes #4669 (#7964)

lucasfernog committed:
fix(bundler): lint and cleanup for #7964 (#8275)

amrbashir committed:
feat: expose `Manager::resources_table` (#8276)

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>

lucasfernog committed:
feat: update to latest wry (#8213)

Co-authored-by: amrbashir <amr.bashir2015@gmail.com>

lucasfernog committed:
fix(ci): skip installing unnecessary deps (#8277)

lucasfernog committed:
fix(ci): skip installing unnecessary deps (#8277)

lucasfernog-crabnebula committed:
feat(core): add AssetManager::iter (#8288)

This new function allows users to iterate on all embedded assets, important if you want to AssetManager::get an asset you are not sure exists.

deid84 committed:
chore: remove unused import `muda::AboutMetadataBuilder` (#8294)

deid84 committed:
chore: aligned `cocoa` crate to 0.25 (#8297)

lucasfernog-crabnebula committed:
feat(core): add tracing for vital functionality closes #5204 (#8289)

* feat(core): add tracing for vital functionality

* Update core/tauri-runtime-wry/src/lib.rs [skip ci]

* Update Cargo.toml [skip ci]

* tracing feature

* wry 0.24.6

* add change tag

* add tracing to CI test

* enhance spans for update check

* remove app from debug impl

cijiugechu committed:
refactor: replace `once_cell` with `std::sync::OnceLock` (#8309)

Our MSRV is currently 1.70 and these new types were introduced in 1.70

ChaseKnowlden committed:
fix: Fix archived links, Closes #8302 (#8315)

i-c-b committed:
fix(bundler): migrate WebView2 offline installer to shorturl (#8292)

FabianLars committed:
chore: Cleanup Readme. Add Sponsor section. (#8269)

* chore: Cleanup Readme. Add Sponsor section.

* move logo into sponsors subfolder

* Update README.md

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>

* Update README.md

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>

* Update README.md

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>

* move getting started up

* remove docs info

---------

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>

Beanow committed:
perf: skip large `Window` and `AppHandle` fields on traces (#8318)

* perf: Skip large Window and AppHandle fields on traces

These contain large fields like image buffers, causing spans/events to be very large when serialized.

Especially the `window::emit::eval` one which is in a hot code path.

* fix: MSRV issues

github-actions[bot] committed:
Apply Version Updates From Current Changes (v1) (#8203)

* apply version updates

* fmt

---------

Co-authored-by: amrbashir <amrbashir@users.noreply.github.com>
Co-authored-by: amrbashir <amr.bashir2015@gmail.com>

amrbashir committed:
chore: manually bump `tauri-utils`

amrbashir committed:
chore: update tauri-cli lock file

zacharyftw committed:
feat(cli): improve error message with invalid target for `build` (#8321)

notriddle committed:
fix: fix incorrect predefined menu item for Close Window (#8328)

* Fix incorrect predefined menu item for Close Window

* change file

---------

Co-authored-by: Amr Bashir <amr.bashir2015@gmail.com>

Hacksore committed:
feat(cli): expose CLI signer flags as environment variables (#8327)

* feat(cli): expose signer flags as environment variables

* Update .changes/cli-signer-env-vars.md

amrbashir committed:
fix(bundler): unset `NSISDIR` and `NSISCONFDIR` when running `makensis` (#8376)

closes #8359

amrbashir committed:
ci: fix publish `@tauri-apps/api` to `latest` instead of `next` (#8371)

* ci: fix publish `@tauri-apps/api` to `latest` instead of `next`

closes #8335

* just remove tag

amr-crabnebula committed:
fix(bundler/nsis): only kill processes of current user (#8390)

* fix(bundler/nsis): only kill processes of current user

Co-authored-by: FabianLars-crabnebula <fabianlars@crabnebula.dev>

* change file

---------

Co-authored-by: FabianLars-crabnebula <fabianlars@crabnebula.dev>
Co-authored-by: Amr Bashir <amr.bashir2015@gmail.com>

github-actions[bot] committed:
Apply Version Updates From Current Changes (v1) (#8377)

Co-authored-by: amrbashir <amrbashir@users.noreply.github.com>

amrbashir committed:
chore(cli/help): mention the need for a second `--` in `tauri dev` (#8388)

* chore(cli/help): mention the need for a second `--` in `tauri dev`

ref: https://github.com/tauri-apps/tauri/issues/8382#issuecomment-1854016310

* add example to clarify [skip ci]

---------

Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

guillemcordoba committed:
Fixed channel deserialisation in iOS (#8386)

* Fixed channel

* Change file

vdang-crabnebula committed:
feat(core): allow the `MockRuntime` to call a command and get a return value  + fix doctest (#8380)

* Add `get_ipc_response` to the `MockRuntime`

* Fix `MockRuntime` doctests

* Add support for raw ipc

* cleanup, add change file

* only a single get fn

---------

Co-authored-by: Lucas Nogueira <lucas@crabnebula.dev>

vdang-crabnebula committed:
feat(core): add test::test::get_ipc_response, closes #8199 (#8228)

* feat(core): Add additional functions to the

* Add documentation and cleanup the code

* Improve and add tests to helper functions

* Clean unecessary code and correct tests

* Make `Ipc` and `IpcKey` public

* Open `Ipc` with public functions

* Update core/tauri/src/test/mod.rs

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>

* cleanup, change file

---------

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>
Co-authored-by: Lucas Nogueira <lucas@crabnebula.dev>

amrbashir committed:
refactor(api): generate types next to modules (#8391)

* refactor(api): generate types next to modules"

this fixes an issue with `moduleResolution: node`

* change file

* Update api-moduleresolution-node.md

amrbashir committed:
refactor(api): generate types next to modules (#8392)

* refactor(api): generate types next to modules"

this fixes an issue with `moduleResolution: node`

* change file

* Update api-module-resolution-node.md

lucasfernog committed:
refactor(clipboard): use arboard instead of tao closes #8177 (#8394)

* refactor(clipboard): use arboard instead of tao closes #8177

* update api lock

* add change file

amrbashir committed:
chore: cleanup issue and PR templates (#8410)

* chore: cleanup issue and PR templates

* Delete .github/config.yml

* Update bug_report.yml

* Update .github/PULL_REQUEST_TEMPLATE.md

Co-authored-by: Fabian-Lars <fabianlars@fabianlars.de>

* Update PULL_REQUEST_TEMPLATE.md

---------

Co-authored-by: Fabian-Lars <fabianlars@fabianlars.de>

amrbashir committed:
ci: fix msrv tests (#8409)

* ci: remove `is-terminal` from crate pinning step

* pin cross-beam

* crossbeam-utils

* image

* linux-raw-sys

* fix version

* comon

* ha?

* tar

* now?

* rustix

* kill me, kill me now

* tar first???

* is-terminal

* I am dying

* remove rustix

* image

amrbashir committed:
feat(api): expose `Resource` class (#8370)

* feat(api): expose `Resource` class

continuation of https://github.com/tauri-apps/tauri/pull/8276

* Apply suggestions from code review

* fmt

---------

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>
Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

amrbashir committed:
ci: access cli release id directly (#8397)

amrbashir committed:
ci: access cli release id directly (#8398)

amrbashir committed:
fix(utils/config): fix parsing `f64` values, closes #8252 (#8407)

dependabot[bot] committed:
chore(deps-dev): bump vite from 4.4.9 to 4.4.12 in /examples/api (#8342)

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 4.4.9 to 4.4.12.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v4.4.12/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v4.4.12/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

lucasfernog committed:
feat(core): set CORS headers on protocol errors (#8419)

* feat(core): set CORS headers on protocol errors

This ensures the frontend can read the error message instead of just showing a CORS error

* fix statuscode

amrbashir committed:
fix(core/dialog): reconstruct path using appropriate separator, closes #8074 (#8408)

* fix(core/dialog): reconstruct path using appropraite separator, closes #8074

* Update dialog-window-forward-slash.md

* Update dialog.rs

* Update core/tauri/src/endpoints/dialog.rs

---------

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>

lucasfernog committed:
feat(core): expose scope::fs::Scope::new (#8432)

lucasfernog committed:
fix(core): scope checks on Android (#8420)

* fix(core): scope checks on Android

On Android, when we call canonicalize() on "/data/user/0/appid" (which is the data dir), the result is a "/data/data/appid" path, so we need to adjust our scope for that.

* clarify code

* apply the logic to all targets

github-actions[bot] committed:
Apply Version Updates From Current Changes (#8274)

Co-authored-by: lucasfernog <lucasfernog@users.noreply.github.com>

lucasfernog committed:
fix: bump tauri-runtime-wry for #8213

lucasfernog committed:
chore(utils): add change file for #7964

lucasfernog committed:
Merge branch '1.x' into dev

amrbashir committed:
refactor!: remove `GlobalWindowEvent` type (#8430)

* refactor!: remove `GlobalWindowEvent` type

* takee references instead

lucasfernog-crabnebula committed:
Merge remote-tracking branch 'upstream/dev' into feat/core-tracing

renovate[bot] committed:
chore(deps) Update dependency eslint-plugin-security to v2 (dev) (#8440)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

FabianLars committed:
fix(cli): Handle target triples with 4 components (#8448)

* fix(cli): Handle target triples with 4 components

Follow up of #8321 which broke the `TAURI_ENV_` vars for linux, windows and ios-sim

* use host for platform as is

* clippy

lucasfernog-crabnebula committed:
ipc spans adjustments

lucasfernog-crabnebula committed:
merge from dev

lucasfernog-crabnebula committed:
delete change file

lucasfernog committed:
fix(cli): remove test only relevant to v1 branch, adjust migration test (#8451)

lucasfernog-crabnebula committed:
improve how request and response are added as values (serialize)

lucasfernog-crabnebula committed:
do not run evalScript sync on android :( freezes the app

olivierlemasle committed:
feat(bundler): Add RPM packaging, closes #4402 (#5202)

* feat(bundler): Add RPM packaging

* feat(bundler): Update 'rpm' to 0.13.1

* Fix fmt

amrbashir committed:
ci: use default options for repository-dispatch (#8455)

jrouaix committed:
rename appimage.sh => build_appimage.sh in error (#8295)

lucasfernog committed:
fix(macos): allow canceling `data-tauri-drag-region` maximization #8312 (#8487)

amrbashir committed:
fix(runtime-wry): propagate wry file drop event to tao event loop, closes #8206 (#8393)

* fix(runtime-wry): propagate wry file drop event to tao event loop

* fix build

* always block default OS file drop behavior

* Update .changes/runevent-filedrop.md [skip ci]

---------

Co-authored-by: Lucas Nogueira <lucas@tauri.studio>
Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>

amrbashir committed:
refactor!: match target triple for `TAURI_ENV_ARCH` (#8486)

* refactor!: match target triple for `TAURI_ENV_ARCH`

* fix build

* Update .changes/cli-hooks-env-vars-breaking.md

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>

* Update tooling/cli/ENVIRONMENT_VARIABLES.md

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>

---------

Co-authored-by: Lucas Fernandes Nogueira <lucas@tauri.app>

repalash committed:
feat: Add files field in macos build config, closes #3290 (#7798)

* Add appContents field in macos tauri config.

* Change MacConfig::appContents to MacConfig::files to make it similar to DebConfig::files.

* Change appContents to files in helloworld/tauri.conf.json

* use common::copy_dir helper

* add change files [skip ci]

---------

Co-authored-by: Lucas Nogueira <lucas@tauri.app>

pashokitsme committed:
feat: add `WindowBuilder::on_download`, closes #8157 (#8159)

* on_download_started & on_download_completed setters

* macos: default handler fn if not set

* remove default macos handler

* doc comments

* unify hooks, change files

---------

Co-authored-by: Lucas Nogueira <lucas@tauri.studio>

lucasfernog-crabnebula committed:
wry 0.35.2

lucasfernog committed:
feat(cli): migrate tauri-build version (#8529)

* feat(cli): migrate tauri-build version

This changes the migrate command to also migrate the tauri-build dependency to v2, previously it only updated the tauri dependency.

* lint

FabianLars committed:
fix(api.js): mock `__TAURI_INTERNALS__` in `mockIPC` and `mockWindows` (#8534)

* fix(api.js): mock `__TAURI_INTERNALS__` in `mockIPC` and `mockWindows`

* mock internals.invoke instead of internals.ipc

FabianLars committed:
fix(examples): Update web example to latest dialog plugin (#8567)

lucasfernog-crabnebula committed:
add change file

lucasfernog-crabnebula committed:
Merge branch 'dev' into feat/core-tracing
